File: task-03/Go/Sub-Task3.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	// Create a reader to read input from the user
	reader := bufio.NewReader(os.Stdin)
	fmt.Print("Enter the number of rows for the diamond (n): ")

	// Read the input as a string
	input, err := reader.ReadString('\n')
	if err != nil {
		fmt.Println("Error reading input:", err)
		return
	}

	// Trim any whitespace and convert to integer
	input = strings.TrimSpace(input)
	n, err := strconv.Atoi(input)
	if err != nil || n <= 0 {
		fmt.Println("Please enter a valid positive integer.")
		return
	}

	// Generate the upper half of the diamond
	for i := 1; i <= n; i++ {
		// Print spaces
		for j := i; j < n; j++ {
			fmt.Print(" ")
		}
		// Print asterisks
		for k := 1; k <= (2*i-1); k++ {
			fmt.Print("*")
		}
		// Move to the next line
		fmt.Println()
	}

	// Generate the lower half of the diamond
	for i := n - 1; i >= 1; i-- {
		// Print spaces
		for j := n; j > i; j-- {
			fmt.Print(" ")
		}
		// Print asterisks
		for k := 1; k <= (2*i-1); k++ {
			fmt.Print("*")
		}
		// Move to the next line
		fmt.Println()
	}
}

File: task-03/Rust/Sub-Task3.rs
use std::io::{self, Write};

fn main() {
    // Prompt the user for input
    print!("Enter the number of rows for the diamond pattern (n): ");
    io::stdout().flush().expect("Failed to flush stdout");

    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read input");

    let n: usize = input.trim().parse().expect("Please enter a valid number");

    // Generate and print the diamond pattern
    print_diamond(n);
}

// Function to generate and print the diamond pattern
fn print_diamond(n: usize) {
    // Print the upper half of the diamond, including the middle row
    for i in 1..=n {
        // Print leading spaces
        for _ in 0..(n - i) {
            print!(" ");
        }
        // Print asterisks
        for _ in 0..(2 * i - 1) {
            print!("*");
        }
        // Move to the next line
        println!();
    }

    // Print the lower half of the diamond
    for i in (1..n).rev() {
        // Print leading spaces
        for _ in 0..(n - i) {
            print!(" ");
        }
        // Print asterisks
        for _ in 0..(2 * i - 1) {
            print!("*");
        }
        // Move to the next line
        println!();
    }
}
